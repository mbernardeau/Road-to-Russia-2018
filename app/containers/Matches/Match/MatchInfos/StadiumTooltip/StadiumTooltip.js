import React from 'react';

import PropTypes from 'prop-types';

const textContainerStyles = {
  display: 'flex',
  color: 'white',
  textShadow: '0 1px 0 rgba(0,0,0,.2)',
  justifyContent: 'space-around',
};

const StadiumTooltip = ({ stadium }) => (
  <div style={{ width: 200 }}>
    <div style={textContainerStyles}>
      <div style={{ textAlign: 'center' }}>
        { stadium.name }
      </div>
      <div style={{ textAlign: 'center' }}>•</div>
      <div style={{ textAlign: 'center' }}>
        { stadium.city }
      </div>
    </div>
    {stadium.photo && <img style={{ width: '100%' }} src={stadium.photo.url} alt={stadium.name} />}
    <div style={textContainerStyles}>
      <div>Capacité</div>
      <div>{stadium.capacity.toLocaleString('fr')} places</div>
    </div>
  </div>
);

StadiumTooltip.propTypes = {
  stadium: PropTypes.shape({
    name: PropTypes.string.isRequired,
    city: PropTypes.string.isRequired,
    photo: PropTypes.shape({
      url: PropTypes.string.isRequired,
      credit: PropTypes.string,
    }),
    capacity: PropTypes.number.isRequired,
  }).isRequired,
};


export default StadiumTooltip;
