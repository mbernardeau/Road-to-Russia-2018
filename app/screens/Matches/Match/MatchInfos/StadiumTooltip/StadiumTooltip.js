import React from 'react'

import PropTypes from 'prop-types'

const StadiumTooltip = ({ name, photo, city, capacity }) => (
  <div style={styles.container}>
    <div style={styles.textContainer}>
      <div style={styles.stadiumText}>{name}</div>
      <div style={styles.stadiumText}>•</div>
      <div style={styles.stadiumText}>{city}</div>
    </div>
    {photo && <img style={styles.img} src={photo.url} alt={name} />}
    <div style={styles.textContainer}>
      <div>Capacité</div>
      <div>{capacity.toLocaleString('fr')} places</div>
    </div>
  </div>
)

const styles = {
  container: {
    width: 200,
  },

  textContainer: {
    display: 'flex',
    color: 'white',
    textShadow: '0 1px 0 rgba(0,0,0,.2)',
    justifyContent: 'space-around',
  },

  stadiumText: {
    textAlign: 'center',
  },

  img: {
    width: '100%',
  },
}

StadiumTooltip.defaultProps = {
  capacity: 0,
  name: '',
  city: '',
}

StadiumTooltip.propTypes = {
  name: PropTypes.string,
  city: PropTypes.string,
  photo: PropTypes.shape({
    url: PropTypes.string.isRequired,
    credit: PropTypes.string,
  }),
  capacity: PropTypes.number,
}

export default StadiumTooltip
