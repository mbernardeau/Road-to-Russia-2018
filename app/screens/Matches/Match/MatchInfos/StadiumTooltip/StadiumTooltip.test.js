import React from 'react'
import renderer from 'react-test-renderer'

import StadiumTooltip from './StadiumTooltip'

describe('screens/Match/StadiumTooltip', () => {
  const stadium = {
    capacity: 69500,
    city: 'Saint-Pétersbourg',
    name: 'Stade Krestovski',
  }

  const stadiumWithPhoto = {
    ...stadium,
    photo: {
      credit: 'Andrew Shiva / Wikipedia, via Wikimedia Commons',
      url:
        'https://upload.wikimedia.org/wikipedia/commons/thumb/7/74/RUS-2016-Aerial-SPB-Krestovsky_Stadium_02.jpg/512px-RUS-2016-Aerial-SPB-Krestovsky_Stadium_02.jpg',
    },
  }

  // Mock toLocaleString because of inconsistant behaviour in node (it works correctly in browsers)
  // eslint-disable-next-line
  Number.prototype.toLocaleString = function() {
    return `${this}`
  }

  it('renders correctly', () => {
    const tree = renderer.create(<StadiumTooltip {...stadiumWithPhoto} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('render with no image if not provided', () => {
    const tree = renderer.create(<StadiumTooltip {...stadium} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
