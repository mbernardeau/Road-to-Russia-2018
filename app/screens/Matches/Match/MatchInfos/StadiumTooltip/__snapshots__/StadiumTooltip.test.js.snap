// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/Match/StadiumTooltip render with no image if not provided 1`] = `
<div
  className="stadium-tooltip"
>
  <div
    className="stadium-tooltip-title"
  >
    <div>
      Stade Krestovski
    </div>
    <div>
      •
    </div>
    <div>
      Saint-Pétersbourg
    </div>
  </div>
  <div
    className="stadium-tooltip-title"
  >
    <div>
      Capacité
    </div>
    <div>
      69500
       places
    </div>
  </div>
</div>
`;

exports[`screens/Match/StadiumTooltip renders correctly 1`] = `
<div
  className="stadium-tooltip"
>
  <div
    className="stadium-tooltip-title"
  >
    <div>
      Stade Krestovski
    </div>
    <div>
      •
    </div>
    <div>
      Saint-Pétersbourg
    </div>
  </div>
  <img
    alt="Stade Krestovski"
    className="stadium-tooltip-photo"
    src="https://upload.wikimedia.org/wikipedia/commons/thumb/7/74/RUS-2016-Aerial-SPB-Krestovsky_Stadium_02.jpg/512px-RUS-2016-Aerial-SPB-Krestovsky_Stadium_02.jpg"
  />
  <div
    className="stadium-tooltip-title"
  >
    <div>
      Capacité
    </div>
    <div>
      69500
       places
    </div>
  </div>
</div>
`;
